mongodb-protolib/src/main/java/com/streamsets/pipeline/stage/origin/mongodb/MongoDBSource.java

```
@@ -192,55 +204,64 @@ private String parseSourceOffset(Document doc, String[] keys, int i) throws Stag
     return parseSourceOffset((Document)doc.get(keys[i]), keys, i+1);
   }
 
-  private void prepareCursor(int maxBatchSize, String offsetField, String lastSourceOffset) {
+  private void prepareCursor(int maxBatchSize, String offsetField, String lastSourceOffset)  {
     String stringOffset = "";
     ObjectId objectIdOffset = null;
+    Date dateOffset = null;
     if (null == cursor) {
       if (null == lastSourceOffset || lastSourceOffset.isEmpty()) {
         objectIdOffset = initialObjectId;
         stringOffset = initialId;
+        dateOffset = initialDate;
       } else {
         if (configBean.offsetType == OffsetFieldType.STRING)
           stringOffset = lastSourceOffset;
-        else
+        else if (configBean.offsetType == OffsetFieldType.OBJECTID)
           objectIdOffset = new ObjectId(lastSourceOffset);
+        else if (configBean.offsetType == OffsetFieldType.DATE) {
+          try {
+            dateOffset = dateFormatter.parse(lastSourceOffset);
+          } catch (ParseException ex) {
+            LOG.error("Could not parse lastSourceOffset to date: {}", lastSourceOffset);
```

Are we intentionally swallowing the exception here? Should we create a StageException instead and wrap the original ParseException?